src/experiments.ts

```diff
@@ -0,0 +1,182 @@
+import { configstore } from "./configstore";
+import * as clc from "colorette";
+import { FirebaseError } from "./error";
+
+export interface Experiment<Name extends string> {
+  name: Name;
+  shortDescription: string;
+  fullDescription?: string;
+  public?: boolean;
+  docsUri?: string;
+  default?: boolean;
+}
+
+function experiment<Name extends string>(
+  name: Name,
+  shortDescription: string,
+  opts?: {
+    fullDescription?: string;
+    public?: boolean;
+    docsUri?: string;
+    default?: boolean;
+  }
+): Experiment<Name> {
+  return {
+    ...opts,
+    name,
+    shortDescription,
+  } as Experiment<Name>;
+}
+
+export const EXPERIMENTS = Object.freeze([
+  // meta:
+  experiment("experiments", "enables the experiments family of commands"),
+
+  // Realtime Database experiments
+  experiment("rtdbrules", "TODO"),
+  experiment("rtdbmanagement", "TODO"),
+
+  // Extensions experiments
+  experiment("ext", "Enables the 'ext:sources:create' command"),
```

Comment:
@pavelgj Can you reword or approve these commands?

1. Do you like the short description?
2. Do you want this to show up in `firebase experiments:list` or just in `firebase experiments:describe`?
2. Do you want to provide a full description and/or docs URL for customers?

Reply:
Resolved offline